modules/cluster-logging-configuration-of-json-log-data-for-default-elasticsearch.adoc

```diff
@@ -0,0 +1,111 @@
+[id="cluster-logging-configuration-of-json-log-data-for-default-elasticsearch_{context}"]
+= Configuration of JSON log data for default Elasticsearch
+
+With the default Elasticsearch log store, you must create an index for each format if the JSON log entries _have different formats_.
+
+[IMPORTANT]
+====
+You must create a separate index for each different JSON log format. Otherwise, forwarding different formats to the same index can cause type conflicts and cardinality problems.
+====
+
+To provide a different index for each format, you configure the `ClusterLogForwarder` custom resource. You use a structure type to construct the index name from a category, Kubernetes label, or input selector.
+
+.Structure types
+
+The `ClusterLogForwarder` provides the following structure types for use with the default Elasticsearch log store:
+
+* `structuredTypeKey` (string, optional) is the name of a message field. The value of that field, if present, is used to construct the index name.
+** `kubernetes.labels.<key>` is the Kubernetes Pod label whose value is used to construct the index name.
+** `openshift.labels.<key>` is the `pipeline.label.<key>` element in the `ClusterLogForwarder` whose value is used to construct the index name.
+** `kubernetes.container_name` uses the container name to construct the index name.
+* `structuredTypeName`: (string, optional) If `structuredTypeKey` is not set or its key is not present, OpenShift Logging uses the value of `structuredTypeName` as the structured type. When you use both `structuredTypeKey` and `structuredTypeName` together,  `structuredTypeName` provides a fallback index name if the key in `structuredTypeKey` is missing from the JSON log data.
+
+[NOTE]
+====
+Although you can set the value of `structuredTypeKey` to any field shown in the "Log Record Fields" topic, the most useful fields are shown in the preceding list of structure types.
+====
+
+.A structuredTypeKey: kubernetes.labels.<key> example
+
+Suppose the following:
+
+* Your cluster is running application pods that produce JSON logs in two different formats, "apache" and "google".
+* The user labels these application pods with `logFormat=apache` and `logFormat=google`.
+* You use the following snippet in your `ClusterLogForwarder` custom resource YAML file.
+
+[source,yaml]
+----
+outputDefaults:
+- elasticsearch:
+    structuredTypeKey: kubernetes.labels.logFormat <1>
+    structuredTypeName: nologformat
+pipelines:
+- inputRefs: <application>
+  outputRefs: default
+  parse: json <2>
+----
+<1> Uses the value of the key-value pair that is formed by the Kubernetes `logFormat` label.
+<2> Enables JSON log forwarding.
+
+In that case, the following structured log record goes to the `app-apache-write` index:
+
+[source,json]
+----
+{
+  "structured":{"name":"fred","home":"bedrock"},
+  "kubernetes":{"labels":{"logFormat": "apache", ...}}
```

Comment:
The ellipsis is highlighted in red in the output.  Not sure if you might want to remove it?

Reply:
The ellipsis indicates unspecified additional elements. It is important for the example, but turns red because it is invalid JSON. I have removed `json` from `[source,json]`.